### DISCUSSION TITLE
Install using pipx

### DISCUSSION THREAD

**Discussion by lamyergeier** [beetbox/beets#5031] [Q&A] [answered]
Pipx creates a venv  and installs beats in it. Will this work with plugins? or is it not recommended to use pip?

Refer: [[pypa/pipx: Install and Run Python Applications in Isolated Environments](https://github.com/pypa/pipx)](https://github.com/pypa/pipx)

```
pipx install beets
```

**Comment by DjSlash** [comment]
> Will this work with plugins?

Yes it will.

> or is it not recommended to use pip?

It's the recommended way to [run the latest source version of beets](https://beets.readthedocs.io/en/stable/faq.html#run-the-latest-source-version-of-beets).

**Comment by RollingStar** [comment]
Is anyone using Poetry? https://python-poetry.org/docs/

**Reply by DjSlash** [reply]
Hi! Please start a new topic for this.

**Comment by ktetzlaff** [comment, marked as answer]
Here's an example of installing *beets* and the 3rd-party plugin *beets-extrafiles* (from git) using *pipx*:
``` sh
pipx install 'beets[lyrics,web]'
pipx inject beets 'git+https://github.com/Holzhaus/beets-extrafiles.git@master'
```
List the *core plugins* you plan to use as *extras* (the `[...]` list of the *pipx install* command) to get plugin dependencies installed automatically (see `extras_require={...}` in [setup.py](https://github.com/beetbox/beets/blob/cc941df025d288d4bdf65967ef658103d15086a1/setup.py#L103) for available *extras*).

To also install *beets* from git, use:
``` sh
pipx install 'beets[lyrics,web] @ git+https://github.com/beetbox/beets.git@master'
pipx inject beets 'git+https://github.com/Holzhaus/beets-extrafiles.git@master'
```
